File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        currentsum = nums[0];
        maxsum = nums[0];
        for i in range(1,len(nums)):
            currentsum = max(currentsum+nums[i],nums[i])
            maxsum = max(maxsum,currentsum)
        return maxsum

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum_ = nums[0];
        int maxsum = nums[0];
        for (int i = 1; i < nums.length; i++){
            sum_ = Math.max(nums[i],sum_+nums[i]);
            maxsum = Math.max(maxsum,sum_);
        }
        return maxsum;

        // int sum = nums[0];
        // int maxsum = nums[0];
        // for (int i = 0; i < nums.length; i++){

        // }
        // for (int j = 1; j<nums.length;j++){
        //     sum
        // }
    }
}